src/opendr/perception/object_detection_2d/nanodet/nanodet_learner.py

```diff
@@ -676,36 +794,59 @@ def infer(self, input, conf_threshold=0.35, iou_threshold=0.6, nms_max_num=100):
         :param iou_threshold: iou threshold
         :type iou_threshold: float, optional
         :param nms_max_num: determines the maximum number of bounding boxes that will be retained following the nms.
-        :type nms_max_num: int
+        :type nms_max_num: int, optional
+        :param hf: determines if model precision.
+        :type hf: bool, optional
+        :param dynamic: determines if the model runs with dynamic input, it can be used in Nanodet Plus head with
+         legacy_post_process=False.
+        :type dynamic: bool, optional
+        :param ch_l: determines if inference will run in channel last format.
+        :type ch_l: bool, optional, optional
         :return: list of bounding boxes of last image of input or last frame of the video
         :rtype: opendr.engine.target.BoundingBoxList
         """
+
+        ch_l = ch_l and self.jit_model is not None
         if not self.predictor:
             self.predictor = Predictor(self.cfg, self.model, device=self.device, conf_thresh=conf_threshold,
-                                       iou_thresh=iou_threshold, nms_max_num=nms_max_num)
+                                       iou_thresh=iou_threshold, nms_max_num=nms_max_num, hf=hf, dynamic=dynamic,
+                                       ch_l=ch_l)
 
         if not isinstance(input, Image):
             input = Image(input)
         _input = input.opencv()
 
         _input, *metadata = self.predictor.preprocessing(_input)
 
-        if self.ort_session:
-            if self.jit_model:
+        if self.trt_model:
+            if self.jit_model or self.ort_session:
                 warnings.warn(
-                    "Warning: Both JIT and ONNX models are initialized, inference will run in ONNX mode by default.\n"
-                    "To run in JIT please delete the self.ort_session like: detector.ort_session = None.")
-            preds = self.ort_session.run(['output'], {'data': _input.cpu().detach().numpy()})
-            res = self.predictor.postprocessing(torch.from_numpy(preds[0]), _input, *metadata)
+                    "Warning: More than one optimizations are initialized, inference will run in TensorRT mode by default.\n"
+                    "To run in a specific optimization please delete the self.ort_session, self.jit_model or "
+                    "self.trt_model like: detector.ort_session = None.")
+            preds = self.trt_model(_input)
+            res = self.predictor.postprocessing(preds, _input, *metadata)
         elif self.jit_model:
-            res = self.jit_model(_input, *metadata)
+            if self.ort_session:
+                warnings.warn(
+                    "Warning: Both JIT and ONNX models are initialized, inference will run in JIT mode by default.\n"
+                    "To run in JIT please delete the self.jit_model like: detector.ort_session = None.")
```

Comment:
Maybe you meant delete the self.ort_session, meaning the onnx session, so JIT model is used?

```suggestion
                    "To run in JIT please delete the self.ort_session like: detector.ort_session = None.")
```